ex_app/lib/main.py

```diff
@@ -127,44 +135,46 @@ def background_thread_task():
                 httpx.PoolTimeout,
         ) as e:
             log(nc, LogLvl.DEBUG, "Ignored error during task polling")
-            sleep(2)
+            await asyncio.sleep(2)
             continue
 
         task = response["task"]
         log(nc, LogLvl.INFO, 'New Task incoming')
         log(nc, LogLvl.DEBUG, str(task))
         log(nc, LogLvl.INFO, str({'input': task['input']['input'], 'confirmation': task['input']['confirmation'], 'conversation_token': '<skipped>'}))
-
+        asyncio.create_task(handle_task(task, nc))
+        await asyncio.sleep(5)
+
+
+async def handle_task(task, nc: NextcloudApp):
+    try:
+        nextcloud = NextcloudApp()
+        if task['userId']:
+            nextcloud.set_user(task['userId'])
+        output = await react(task, nextcloud)
```

Comment:
This might not works as smoothly as it looks, because most of our tools still do synchronous HTTP requests, I think. But perhaps lets cahnge that in a different PR.